ddprof-lib/src/main/cpp/os_macos.cpp

```diff
@@ -216,6 +216,7 @@ SigAction OS::replaceSigsegvHandler(SigAction action) {
   sigaction(SIGSEGV, NULL, &sa);
   SigAction old_action = sa.sa_sigaction;
   sa.sa_sigaction = action;
+  sa.sa_flags = SA_SIGINFO | SA_ONSTACK;
```

Comment:
What is the memory impact of doing this with the extensive usage of `SafeAccess` as well as segfaults originating in the JVM? How do signal handlers with different flags compose?

Reply:
Memory impact is about 1KB per invocation required on stack.
The signal handlers are processed one by one so the flags are not composed/merged and for each handler the information to be delivered is prepared according to the flags.

I would say that the extra overhead should be minimal - even with safe access we are trying to avoid accessing known invalid locations so getting sigseg and entering this handler should be rather rare.